spec.bs

```diff
@@ -1109,10 +1131,8 @@ The <dfn attribute for="AppHistoryDestination">sameDocument</dfn> getter steps a
 <div algorithm>
   The <dfn>inner `navigate` event firing algorithm</dfn> is the following steps, given an {{AppHistory}} |appHistory|, an {{AppHistoryNavigationType}} |navigationType|, an {{AppHistoryNavigateEvent}} |event|, an {{AppHistoryDestination}} |destination|, a [=user navigation involvement=] |userInvolvement|, and a [=list=] of {{FormData}} [=FormData/entries=] or null |formDataEntryList|:
 
-  1. [=AppHistory/Promote the upcoming non-traverse navigation to ongoing=] given |appHistory|.
-  1. Let |ongoingNavigation| be null.
-  1. If |destination|'s [=AppHistoryDestination/key=] is null, then set |ongoingNavigation| to |appHistory|'s [=AppHistory/ongoing non-traverse navigation=].
-  1. Otherwise, if |appHistory|'s [=AppHistory/ongoing traverse navigations=][|destination|'s [=AppHistoryDestination/key=]] [=map/exists=], then set |ongoingNavigation| to |appHistory|'s [=AppHistory/ongoing traverse navigations=][|destination|'s [=AppHistoryDestination/key=]].
+  1. [=AppHistory/Promote the upcoming navigation to ongoing=] given |appHistory| and |destination|'s [=AppHistoryDestination/key=].
+  1. Let |ongoingNavigation| be |appHistory|'s [=AppHistory/ongoing navigation=].
```

Comment:
I factored out all these steps into a single helper algorithm, versus the implementation which spreads them out a bit. I am pretty sure they are equivalent (and maybe we could clean up the implementation), but it'd be good to double-check.

Reply:
Agreed, I'll switch to this.